devlib/target.py

```diff
@@ -188,6 +190,10 @@ def connect(self, timeout=None):
         self._update_modules('connected')
         if self.platform.big_core and self.load_default_modules:
             self._install_module(get_module('bl'))
+        if self.busybox is not None:
+            # Busybox is used on this target so pass on to connections
+            # Some connections (i.e. gem5) need to wrap some commands
+            self.conn.set_busybox_wrapper(self.busybox)
```

Comment:
This should not be set here at all if a device has been previously set up, then the `get_installed` call above will not return `None`. If at this point busybox is `None`, then the device has not been set up, and it will be set when `setup()` is called.